frontend/workflows/projectCatalog/src/details/card.tsx

```diff
@@ -0,0 +1,149 @@
+import React from "react";
+import { Card, ClutchError, Error, Grid, styled, Typography } from "@clutch-sh/core";
+import { LinearProgress } from "@material-ui/core";
+
+export type CardTypes = "Dynamic" | "Metadata";
+
+export interface DetailsCard {
+  type: CardTypes;
+}
+
+interface CardTitleProps {
+  title: string;
+  titleIcon?: React.ReactNode;
+  endAdornment?: React.ReactNode;
+}
+
+interface CardBodyProps {
+  children?: React.ReactNode;
+  loading?: boolean;
+  error?: ClutchError;
+}
+
+interface BaseCardProps extends CardTitleProps, CardBodyProps {
+  reloadInterval?: number;
+  autoReload?: boolean;
+  fetchDataFn?: () => Promise<any>;
+  onSuccess?: (data: any) => void;
+  onError?: (error: ClutchError | undefined) => void;
+}
+
+interface CardProps extends DetailsCard, BaseCardProps {}
+
+const StyledCard = styled(Card)({
+  width: "100%",
+  height: "fit-content",
+  padding: "15px",
+});
+
+const StyledProgressContainer = styled("div")({
+  height: "4px",
+  ".MuiLinearProgress-root": {
+    backgroundColor: "rgb(194, 200, 242)",
+  },
+  ".MuiLinearProgress-bar": {
+    backgroundColor: "#3548D4",
+  },
+});
+
+const StyledRow = styled(Grid)({
+  marginBottom: "15px",
+});
+
+const CardTitle = ({ title, titleIcon, endAdornment }: CardTitleProps) => (
+  <>
+    {titleIcon && (
+      <Grid item xs={1}>
+        {titleIcon}
+      </Grid>
+    )}
+    <Grid item xs={8}>
+      <Typography variant="h4">{title}</Typography>
+    </Grid>
+    {endAdornment && (
+      <Grid
+        container
+        item
+        direction="row"
+        xs={3}
+        spacing={1}
+        alignItems="center"
+        justify="flex-end"
+      >
+        {endAdornment}
+      </Grid>
+    )}
+  </>
+);
+
+const CardBody = ({ loading, error, children }: CardBodyProps) => (
+  <>
+    <StyledRow>
+      {loading && (
+        <StyledProgressContainer>
+          <LinearProgress color="secondary" />
+        </StyledProgressContainer>
+      )}
+    </StyledRow>
+    {error ? <Error subject={error} /> : children}
+  </>
+);
+
+const BaseCard = (props: CardProps) => {
+  const [isLoading, setIsLoading] = React.useState<boolean>(false);
+  const [cardError, setCardError] = React.useState<ClutchError | undefined>(undefined);
+  const { children, title, titleIcon, endAdornment, loading, error } = props;
+
+  const fetchData = () => {
+    const { fetchDataFn, onSuccess, onError } = props;
+
+    if (fetchDataFn) {
+      setIsLoading(true);
+
+      fetchDataFn()
+        .then(data => {
+          if (onSuccess) {
+            onSuccess(data);
+          }
+          setCardError(undefined);
+        })
+        .catch(err => {
+          if (onError) {
+            onError(err);
+          }
+          setCardError(err);
+        })
+        .finally(() => setIsLoading(false));
+    }
+  };
+
+  React.useEffect(() => {
+    const { autoReload, reloadInterval = 30000 } = props;
```

Comment:
should autoreload have a default value?

Reply:
I thought booleans defaulted to false generally, I'll add one to be explicit though. Fixed.